### ISSUE TITLE
username_unclaimed can be used to confuse sherlock

### ISSUE THREAD

**Issue by DecryptingElectrons** [sherlock-project/sherlock#299]
While some of the username_unclaimed entries aren't "noonewouldeverusethis7", most of them are. 

It seems it would be possible for that username to exist intentionally or otherwise, causing the program to fail to find usernames for that site.

A simple solution could be a program argument that will replace all username_unclaimed entries with a high entropy string.

**Comment by hoadlck** [comment]
Unfortunately, several sites that did not have noonewouldeverusethis7 now do have it.  Which is ironic.

For the tests to work, we have to really have a name that we are positive that is not used.  A randomly generated name would not do that.

However, it does make me wonder if we were to use a bunch of randomly generated names. Statistically, it would be likely that many of them would be unknown.  But, this would require throwing many tries at the same site, which might cause the sites to get annoyed with Sherlock.

The other problem is that some sites have limits on the username format.  A randomly generate name would be more likely to run into those limits.

**Comment by Beall619** [comment]
So it seems like a solution is, an additional program that is specifically designed for testing & building a good user_unclaimed for each site.

*would require a definition for username symbols requirements per site*

The program appropriately would:

-hit each site with a random 'symbol valid' username until it doesn't match what would appear for username_claimed

-save a new data.json with each random string


The program should probably run over tor by default (while taking longer) to prevent ip bans.

**Comment by hoadlck** [comment]
Yes, something like this would be nice, but it would never be able to be completely automatic.  There is just too much variation in site layout and usernames.

**Comment by not-verified** [comment]
Wouldn't using random usernames of the format [a-z]{10} work on virtually all websites?

Even for websites with 100.000.000 users (all having a username [a-z]{10}), the chance of collision would be smaller than one to a million.

**Comment by mpieters93** [comment]
Wouldn't using random usernames of the format [a-z]{10} work on virtually all websites? 

Even for websites with 100.000.000 users (all having a username [a-z]{10}), the chance of collision would be smaller than one to a million.

**Comment by hoadlck** [comment]
The number of usernames available for a pure alphabetic username consisting of only the 26 letters of the alphabet, would be 
```
26^10 - 100,000,000 = 141,167,095,653,376 - 100,000,000 = 141,166,995,653,376
```
One in 141 trillion.  So, yes, there would be plenty of usernames available.

The problem is that we would need to hit the site with quite a few of them to have some statistical confidence.  And that is not being a good internet citizen: throwing a bunch of hits at a website would look alot like a DOS attack.

**Comment by mpieters93** [comment]
I might be mistaken, but I do not think we would need more than one request. The odds of accidentally hitting a username in use would be roughly  `1e14 / 1e8 = 1,000,000 `.

I think this is not worse than picking a random username to hardcode, as people will (unfortunately) inevitably register that one. The statistical confidence for the random method would be p <= 0.000001 for a false positive, which is not that bad in my opinion.

I'm curious to your thoughts.

**Comment by hoadlck** [comment]
I am not sure if we are talking about the same thing.  Or, maybe I am not understanding what is being proposed.

The unclaimed username is only used for tests.  The test ensures that changes to the site response are detected, so that Sherlock will not issue a false positive.  It has no impact on the normal usage of Sherlock.

The current strategy is to curate a list of both used and unused usernames.  What I thought was being talked about was to ditch the curation of unclaimed usernames and do something random.  The problem I see with that is that one would have to throw many different random usernames at the site to be confident that the positive detection of each of those random names was really because the site changed its behavior....not just that we got unlucky.

I do think that doing something like the above would be much more robust than what we are doing now.  Because, we get failures because of people registering unclaimed usernames.  If we searched a 1000 random usernames in the tests, then we would have a good indication that the site had changed and we were getting a false positive.

**Comment by ppfeister** [comment]
Originally resolved by #1637
